azure-iot-hub/tests/test_iothub_amqp_client.py

```diff
@@ -82,10 +95,69 @@ def test_raise_exception_on_send_fail(self, mocker, mock_uamqp_SendClient):
 
     @pytest.mark.it("Disconnect a Device")
     def test_disconnect_sync(self, mocker, mock_uamqp_SendClient):
-        iothub_amqp_client = IoTHubAmqpClient(
+        iothub_amqp_client = IoTHubAmqpClientSharedAccessKeyAuth(
             fake_hostname, fake_shared_access_key_name, fake_shared_access_key
         )
         amqp_client_obj = mock_uamqp_SendClient.return_value
         iothub_amqp_client.disconnect_sync()
 
         assert amqp_client_obj.close.call_count == 1
+
+
+@pytest.mark.describe("IoTHubAmqpClientTokenAuth - Amqp Client Connections")
+class TestIoTHubAmqpClientTokenAuth(object):
+    @pytest.mark.it("Get Token From TokenCredential Object")
```

Comment:
These should be written as if the sentence begins with "it", i.e. "Gets token from TokenCredential object".

There's a whole theory we have to writing tests where the description for a testcase is a sentence fragment, in indicative-mood third-person present tense, where the subject is implied as the noun defined with pytest.mark.describe. Which sounds complicated, but like I said, it really just means write the test as if the sentence begins with "it".

tl;dr is if you can't write a testname with that grammatical structure, it's a red flag that the test is poorly defined. 

A problem here of course is that the nouns in the pytest.mark.describes are also kind of flawed, but again, that's how they've always been. So IDK, you can probably also just leave all this stuff alone for now so this doesn't turn into a complete refactor of the testfile.

Nontheless, ping me sometime and we can go over some of these conventions for future reference.

Reply:
Yeah, the test file had these issues before, and I simply stuck with the (flawed) format that was already there. I'll probably stick with it the way it is.